package/src/plugins/has-vite-plugin.ts

```diff
@@ -0,0 +1,58 @@
+import type { AstroConfig } from "astro";
+import { PluginOption } from "vite";
+import { definePlugin } from "../core/define-plugin.js";
+import { hasVitePlugin } from "../utilities/has-vite-plugin.js";
+
+function getPlugins(
+	store: Set<
+		Extract<
+			NonNullable<NonNullable<AstroConfig["vite"]["plugins"]>[number]>,
+			{ name: string }
+		>
+	>,
+	plugins: AstroConfig["vite"]["plugins"],
+) {
+	if (plugins) {
+		for (const plugin of plugins) {
+			if (!plugin) continue;
+
+			if (Array.isArray(plugin)) {
+				getPlugins(store, plugin);
+				continue;
+			}
+
+			if (plugin instanceof Promise) {
```

Comment:
1. There's no really no way to get those?
2. Are those frequent? Genuinely curious as I've never seen any

Reply:
There probably is a way, but then wouldn't all the utils/plugins that depend on `hasVitePlugin` have to be asynchronous? I don't think they are frequent, I haven't seen one before